### ISSUE TITLE
gtestp4c infinite loop ?

### ISSUE THREAD

**Issue by jafingerhut** [p4lang/p4c#1900]
On an Ubuntu 18.04 Linux system with latest updates as of 2019-Apr-26, but no other packages installed on it, and `p4c` source code with this version:
```
commit ae6fb228a0d307466f482affd2b47fbd079f3c9e
Author: Mihai Budiu <mbudiu@vmware.com>
Date:   Thu Apr 25 16:46:31 2019 -0700
```
I tried installing behavioral-model, p4c, and the packages that they depend upon, and then running p4c's `cd build ; make check`.  It reached the point in the output shown below of running `./test/gtestp4c`, and then continued using CPU for over 30 minutes without making further progress.
```
[----------] 1 test from FromJSONTest
[ RUN      ] FromJSONTest.load_ir_from_json
```
Later I can add a link to the full build logs on this machine where I tried it, and then try reproducing it with another Ubuntu 18.04 VM from scratch to see if it is reproducible.

**Comment by jafingerhut** [comment]
I have also seen this with same version of p4c on an Ubuntu 16.04 Linux system (also fresh install, also with latest updates as of as of 2019-Apr-26).

Same commit of p4c involved.

I did not see gtestp4c running a long time like this the last time I ran the same install script to install p4c (this one https://github.com/jafingerhut/p4-guide/blob/master/bin/install-p4dev-p4runtime.sh ) when I ran it, then ran `cd build ; make check` on 2019-Apr-21, with a slightly earlier version of behavioral-model and p4c being installed then.
